src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -597,6 +610,7 @@ double DecisionTree<FitnessFunction,
       UseWeights ? weights.subvec(begin, begin + count - 1) : weights);
   size_t bestDim = datasetInfo.Dimensionality(); // This means "no split".
   const size_t end = dimensionSelector.End();
+
   for (size_t i = dimensionSelector.Begin(); i != end;
        i = dimensionSelector.Next())
```

Comment:
In this function, couldn't we just avoid this part of the code by skipping down to the block of code that calculates class probabilities when `maximumDepth == 1`?  That would make the whole changeset a lot simpler in my opinion, since the only place `maximumDepth` would need to be considered would be in this overload and the non-DatasetInfo overload of `Train<UseWeights>()`.

Reply:
Updated as suggested